### PR TITLE
feat: Update proto def to pass 'buf lint'

### PR DESCRIPTION
**Pull request by alilleybrinker**
This commit modifies the protobuf definition for Hipcheck plugins to meet the expectations of 'buf lint', which checks protobuf definitions against a set of best practices. The main goal here is to make the service definition more defensive against forward-compatibility issues.

This also includes updates to the code that interacts with the generated code to make sure it still compiles.

### CODE REVIEW

**Comment by j-lanson** [comment]
I imagine this will break the working communication with the dummy plugin via `hc plugin`

**Comment by alilleybrinker** [comment]
Ah, you're right that I'll need to update that as well. I'll amend the patch with that.

**Comment by alilleybrinker** [comment]
Julian has some in-flight changes to the demo plugin with stuff that we'll be pulling out into our Rust plugin SDK, and I'd rather that get merged first and then rebase this on top. So this'll wait until that's ready.

**Comment by alilleybrinker** [comment]
Resolved merge conflicts, now to get it to compile again haha

File: hipcheck/src/plugin/types.rs [hunk 1]
```diff
@@ -275,34 +331,66 @@ impl TryFrom<Query> for PluginQuery {
...
+/// `query` field on the message can technically be missing.
+///
+/// This case is handled in the `message` method to flatten
+/// that kind of error so code consuming the stream doesn't
+/// have to worry about it.
+type QueryStream = Box<dyn Stream<Item = StdResult<PluginQuery, Status>> + Send + Unpin + 'static>;
+
 impl MultiplexedQueryReceiver {
-	pub fn new(rx: Streaming<PluginQuery>) -> Self {
+	pub fn new(rx: QueryStream) -> Self {
```
Comment (j-lanson) [inline hipcheck/src/plugin/types.rs:359]: Was it changes from `buf lint` that caused the need for this to become `Box<dyn Stream>` or was it a deficiency in my existing code?

Comment (alilleybrinker) [reply hipcheck/src/plugin/types.rs:359]: Good catch! This was necessary for compilation at one point during the refactoring, but I just checked and I was able to remove it without issue in the final version.

**Comment by alilleybrinker** [comment]
@j-lanson removed boxed query stream. Ready for re-review

**Review by j-lanson — APPROVED** [review]
Looks good!

**Pull request merged**